### PR TITLE
Add .git with remote details before clone if needed

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

### CODE REVIEW

File: utils/git.go [hunk 1]
```diff
@@ -361,25 +394,6 @@ func (gm *GitManager) dryRunClone(destination string) error {
...
-		return fmt.Sprintf(bitbucketServerHttpsFormat, gm.git.APIEndpoint, gm.git.RepoOwner, gm.git.RepoName), nil
-	case vcsutils.AzureRepos:
-		azureEndpointWithoutHttps := strings.Join(strings.Split(gm.git.APIEndpoint, "https://")[1:], "")
-		return fmt.Sprintf(azureDevopsHttpsFormat, gm.git.RepoOwner, azureEndpointWithoutHttps, gm.git.Project, gm.git.RepoName), nil
-	default:
-		return "", fmt.Errorf("unsupported version control provider: %s", gm.git.GitProvider.String())
-	}
-}
-
 func toBasicAuth(token, username string) *githttp.BasicAuth {
```
Comment (sverdlov93) [inline utils/git.go:397]:
```suggestion
func toBasicAuth(username, token string) *githttp.BasicAuth {
```

File: utils/git.go [hunk 2]
```diff
@@ -106,35 +161,13 @@ func (gm *GitManager) Clone(destinationPath, branchName string) error {
 	}
 	repo, err := git.PlainClone(destinationPath, false, cloneOptions)
 	if err != nil {
-		return fmt.Errorf("'git clone %s from %s' failed with error: %s", branchName, gitRemoteUrl, err.Error())
+		return fmt.Errorf("'git clone %s from %s' failed with error: %s", branchName, gm.remoteGitUrl, err.Error())
```
Comment (sverdlov93) [inline utils/git.go:164]: 'git clone %s from %s' is not a command so why we wrap it with `'`

File: utils/git.go [hunk 3]
```diff
@@ -64,14 +61,75 @@ type CustomTemplates struct {
...
+	gitRemote, err := gm.repository.Remote(gm.remoteName)
+	if err != nil {
+		return nil, fmt.Errorf("'git remote %s' failed with error: %s", gm.remoteName, err.Error())
+	}
+
+	if len(gitRemote.Config().URLs) < 1 {
+		return nil, errors.New("failed to find git remote URL")
+	}
+
+	gm.remoteGitUrl = gitRemote.Config().URLs[0]
```
Comment (sverdlov93) [inline utils/git.go:103]: what happens if url > 1 ?

File: utils/git.go [hunk 4]
```diff
@@ -64,14 +61,75 @@ type CustomTemplates struct {
...
+
+	if !dotGitExists {
+		// If .git directory doesn't exist, create it with the given remote URL
+		gm.remoteGitUrl = remoteHttpsGitUrl
+		err = vcsutils.CreateDotGitFolderWithRemote(".", vcsutils.RemoteName, remoteHttpsGitUrl)
+		return gm, err
+	}
+
+	if gm.repository == nil {
+		if _, err = gm.SetLocalRepository(); err != nil {
```
Comment (sverdlov93) [inline utils/git.go:89]: can we find another name for this instead of repository?

File: utils/git.go [hunk 5]
```diff
@@ -64,14 +61,75 @@ type CustomTemplates struct {
...
-	return &GitManager{dryRunRepoPath: clonedRepoPath, remoteName: "origin", auth: basicAuth, dryRun: dryRun, customTemplates: templates, git: gitParams}, nil
+
+	if !dotGitExists {
+		// If .git directory doesn't exist, create it with the given remote URL
+		gm.remoteGitUrl = remoteHttpsGitUrl
+		err = vcsutils.CreateDotGitFolderWithRemote(".", vcsutils.RemoteName, remoteHttpsGitUrl)
+		return gm, err
+	}
+
+	if gm.repository == nil {
```
Comment (sverdlov93) [inline utils/git.go:88]: can we find another name for this instead of repository?

File: utils/git.go (hunk 3 above)
Comment (omerzi) [reply utils/git.go:103]: The first one should be the URL from which the repository was cloned. Other URLs might represent the forks, etc.

Comment (omerzi) [reply utils/git.go:103]: @sverdlov93

**Pull request merged**